### PR TITLE
Don't monitor health if single node rolling upgrade in E2E tests

### PR DESCRIPTION
**Pull request by sebgl**
Disable continuous health check monitoring if we're doing a rolling
upgrade of a 1-node cluster.

This test was flappy, since a 1-node cluster upgrade will temporarily be
Red while shards are initializing after the upgrade.

The PR also introduces an optional "Skip" field in the TestStep. If
defined, a function returning True will disable the associated test.

Fixes https://github.com/elastic/cloud-on-k8s/issues/1752.

### CODE REVIEW

File: test/e2e/test/elasticsearch/steps_mutation.go [hunk 1]
```diff
@@ -97,6 +106,21 @@ func (b Builder) MutationTestSteps(k *test.K8sClient) test.StepList {
...
 
+func IsOneNodeRollingUpgrade(b Builder) bool {
+	if b.MutatedFrom == nil {
+		return false
+	}
+	initial := b.MutatedFrom.Elasticsearch
+	mutated := b.Elasticsearch
+	// consider we're in the 1-node rolling upgrade scenario if we mutate
+	// from one node to one node with the same name
+	if len(initial.Spec.Nodes) == 1 && len(mutated.Spec.Nodes) == 1 &&
```
Comment (barkbay) [inline test/e2e/test/elasticsearch/steps_mutation.go]: Should we not rely on the data node count, in the same spirit that https://github.com/elastic/cloud-on-k8s/blob/855643a1cad50037b119f4339e1cdf320682829d/test/e2e/test/elasticsearch/settings.go#L13 ?

Comment (sebgl) [reply test/e2e/test/elasticsearch/steps_mutation.go]: Good point 👍

**Comment by sebgl** [comment]
Jenkins please test this.

**Comment by sebgl** [comment]
Jenkins test this.

**Comment by barkbay** [comment]
jenkins test this please

File: test/e2e/test/step.go [hunk 2]
```diff
@@ -33,6 +34,9 @@ func (l StepList) WithStep(testStep Step) StepList {
 // RunSequential runs the StepList sequentially, and fails fast on first error.
 func (l StepList) RunSequential(t *testing.T) {
 	for _, ts := range l {
+		if ts.Skip != nil && ts.Skip() {
+			continue
```
Comment (david-kow) [inline test/e2e/test/step.go]: Should we log that step was skipped?

**Pull request merged**